app/code/Magento/Sales/Model/Order/Invoice.php

```diff
@@ -493,16 +493,19 @@ public function roundPrice($price, $type = 'regular', $negative = false)
      */
     public function getItemsCollection()
     {
-        if (!$this->hasData(InvoiceInterface::ITEMS)) {
-            $this->setItems($this->_invoiceItemCollectionFactory->create()->setInvoiceFilter($this->getId()));
-
-            if ($this->getId()) {
-                foreach ($this->getItems() as $item) {
```

Comment:
@antonkril what if getItemsCollection would return only the collection as it expected from the name of the method?